packages/iov-lisk/src/derivation.ts

```diff
@@ -25,3 +25,19 @@ export function pubkeyToAddress(pubkey: Uint8Array): Address {
   const addressString = Long.fromBytesLE(firstEightBytes, true).toString(10) + "L";
   return addressString as Address;
 }
+
+export function isValidAddress(address: string): boolean {
+  return isValidAddressWithEnding(address, "L");
+}
+
+export function isValidAddressWithEnding(address: string, ending: string): boolean {
+  const tail = address.slice(-ending.length);
+  if (tail !== ending) {
+    return false;
+  }
+  const addressString = address.slice(0, -ending.length);
+  // parse as unsigned base 10 number and verify re-encoding matches encoding
+  // this is no leading zeros, no decimals, within 0 <= addressString <= 2**64-1
+  const addressNumber = Long.fromString(addressString, true, 10); // true => unsigned
```

Comment:
Long behaves unreliable wenn passing string values > 64 bit. See e.g.

> Long.fromString("18446744073709551616", true).greaterThan(Long.fromString("18446744073709551615", true))
false

So we need to use bn.js

Reply:
I see your point, but since we care about the equality of `parse with long -> encode to string`, we know this only works with integers between 0 and 2**64-1 with no leading zeros and no decimal points. Otherwise the output is somehow different.

Since this is just an equality test, I think Long is fine here.